### ISSUE TITLE
Automate the DelayJS Script Update on plugin

### ISSUE THREAD

**Issue by jeawhanlee** [wp-media/wp-rocket#6369]
**Is your feature request related to a problem? Please describe.**
Whenever we need to update the delayJS on the plugin to a new version, there's quite some manual process involved which creates just enough room for mistakes.
- First we manually copy the unminified script from the branch.
- Paste it in an online tool to minifiy it
- Copy it and update the script on plugin.

**Describe the solution you'd like**
What we can do is to automate this process by running a command from the CLI.
I propose we use a bash script then pass the branch name to be used as an argument
e.g `sudo sh delayjs.sh branch-name`

Breakdown:
- First, We will update the gulp.js file to add specific task to build minified delayjs
- Bash Content process
   - Clone the delayjs repo to a temp dir
   - cd into temp dir
   - Switch to the branch passed as argument
   - Copy the delayjs content into the delayjs file in plugin
   - Run the new gulp task to build unminified delayjs script
   - Delete temp dir

**Describe alternatives you've considered**
Another alternative is to build a minified version of the script on the delayjs branch maybe with gulp, this cut off the gulp step we automate.

**Additional context**
We can also consider updating the related test automatically since it's also just about updating the script in the fixture [here](https://github.com/wp-media/wp-rocket/blob/f471c32f7efda54e7b0ff69ffd2a0acd808a314a/tests/Fixtures/inc/Engine/Optimization/DelayJS/Subscriber/addDelayJsScript.php#L10)

**Comment by gmetais** [comment]
Hi @jeawhanlee,
When you do so, I assume you will use Uglify during compilation. I would like to suggest an Uglify parameter that would produce a lighter result:

```
{
  mangle: {
     properties: {
       regex: /^_/
    }
  }
}
```

`regex: /^_/` means that all the class properties that start with an underscore are renamed (shortened). As all internal properties (and only internal) have an underscore, we are sure we won't generate a bug.

Once the technology is in place, I will duplicate your compilation tooling/config into the DelayJS project, so that automated tests are launched on the minified code too, to prevent regressions.

**Comment by jeawhanlee** [comment]
Many thanks for the heads-up @gmetais 🙏

**Comment by MathieuLamiot** [comment]
As discussed during the daily, to validate this we would just need to compare the script manually imported vs. the one automatically imported and ensure both strings match.
@jeawhanlee has this been done already or not? Otherwise, @wp-media/qa-team can pick it up.

**Comment by jeawhanlee** [comment]
@MathieuLamiot this has been done but uglify compiles these files in a different way from the online tools we use so the minified versions are different, doing a diff check won't do so much, reason I was suggesting if we handled this like any delay js PR update to be sure that the minified version works as expected.

**Comment by MathieuLamiot** [comment]
Ok understood, it makes sense then. @Mai-Saad we would indeed need to run NRT for delayJS with this then 🤔 Is it something doable by devs or maybe @hanna-meda? I am just afraid it is a tough test plan to begin with 😬

**Comment by hanna-meda** [comment]
@MathieuLamiot on my own, I don't know where to start. I need coaching either way

**Comment by gmetais** [comment]
I don't think that full NRT test is required, minification issues are rare and in case they occur, they usually break 100% of functionalities, not just one or two.

I rather suggest launching the automated Playwright tests over the minified version in a Github action. This is something I can work on next week.
